### DISCUSSION TITLE
test icon drop in and load

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [Mariusthvdb/custom-icons#27] [Ideas]
since currently we need to write the path of each icon into the js resource and update all references to those is a bit of a hassle, I hope we can do as other integrations do. Simply have a map with the actual icons, and load those icons. A bit like [Fontawesome](https://github.com/thomasloven/hass-fontawesome/tree/master/custom_components/fontawesome) does, or its clone [simple-icons](https://github.com/vigonotion/hass-simpleicons/tree/master/custom_components/simpleicons).

Ive tried to create an initial js resource, and it loads, but I see a 'forbidden' in inspector...
<img width="821" alt="Schermafbeelding 2022-05-18 om 09 31 11" src="https://user-images.githubusercontent.com/33354141/168983040-fcd753ae-cc9b-4976-8482-df2476541cda.png">

would you know if this could be feasible?
[icon_picker.js.zip](https://github.com/Mariusthvdb/custom-icons/files/8714124/icon_picker.js.zip)

**Comment by Mariusthvdb** [comment]
experimental custom component is working, copied form Fontawesome, and carefully edited the brand out of it, as I only want a way to drop icons and be able to. use them in HA.

**Comment by arallsopp** [comment]
When I had [failures](https://github.com/arallsopp/hass-hue-icons/issues/116) trying to load the icons via JS in the frontend, I created a custom integration version that loads hass-hue-icons from disk. Might be a useful reference?

https://github.com/arallsopp/hass-hue-icons-integration

**Comment by Mariusthvdb** [comment]
how could I have missed that ;-)

still, that requires the icons to be mapped in the main.js file?

my idea is to simply be able to drop an icon.svg in the directory , and then be able to use it in the picker and yaml directly. no more hassle with mapping them and editing paths...

just dropped your adore file and voila:

```
- title: Custom icons
  path: custom_icons
  icon: cli:adore

```
using `cli:` for the custom_component, to differentiate between `cil:` for the plugin resource ...
<img width="475" alt="Schermafbeelding 2022-05-20 om 11 11 15" src="https://user-images.githubusercontent.com/33354141/169495739-e5785e35-4391-40c7-b146-57f4d449f09e.png">

**Reply by arallsopp** [reply]
It predates the picker, I think. But it does let you load svg files directly from the disk, IIRC.  It was a while back now, so I forget quite how it worked :)

The documentation / readme is just a clone of the hass-hue-icons repo at that time, so probably needs to be looked again.

I think the issue will be having the picker be aware of what's in the custom folder, as you maybe don't want to be scanning the disc every time someone types a character in the picker.
